api/apis/app_handler.go

```diff
@@ -219,37 +224,37 @@ func (h *AppHandler) appSetCurrentDropletHandler(authInfo authorization.Info, w
 	appGUID := vars["guid"]
 
 	var payload payloads.AppSetCurrentDroplet
-	rme := h.decoderValidator.DecodeAndValidateJSONPayload(r, &payload)
-	if rme != nil {
-		writeRequestMalformedErrorResponse(w, rme)
+	err := h.decoderValidator.DecodeAndValidateJSONPayload__NewStyle(r, &payload)
+	if err != nil {
+		writeErrorResponse(w, presenter.ForError(err))
 		return
 	}
 
-	app, err := h.appRepo.GetApp(ctx, authInfo, appGUID)
+	app, err := h.appRepo.GetApp__NewStyle(ctx, authInfo, appGUID)
 	if err != nil {
-		h.handleGetAppErr(err, w, appGUID)
+		writeErrorResponse(w, presenter.ForReadError(err))
 		return
 	}
 
 	dropletGUID := payload.Data.GUID
-	droplet, err := h.dropletRepo.GetDroplet(ctx, authInfo, dropletGUID)
+	droplet, err := h.dropletRepo.GetDroplet__NewStyle(ctx, authInfo, dropletGUID)
 	if err != nil {
 		switch err.(type) {
```

Comment:
I hope that this is quite a corner case where would have to use those terrible switches in rest handlers, right? I really hoped that I would not see such a switch again :(

Reply:
+1

Maybe it would make the helper functions ugly, but it does feel like this is a common case that could be shifted to a helper. 

If I'm reading it right, then we got an error while looking something up in order to create or modify something else. In this case, if we get a forbidden or not found, we want to translate that into an "unprocessable entity" error. We will probably need this logic in almost every handler.

Reply:
Yes, that looks much better.